examples/udp-echo.rs

```diff
@@ -0,0 +1,259 @@
+                    continue;
+                }
+
+                // swap src and dst
+                let mut udp_packet = udp_packet?;
+                let src_addr = udp_packet.src_addr;
+                let dst_addr = udp_packet.dst_addr;
+                udp_packet.src_addr = dst_addr;
+                udp_packet.dst_addr = src_addr;
+
+                // send to writer
+                tx.send(udp_packet)?;
+            }
+            Ok::<(), Box<dyn std::error::Error>>(())
+        };
+        if let Err(err) = block() {
+            println!("Reader {}", err);
+        }
+    });
+
+    let writer = std::thread::spawn(move || {
+        let block = || {
+            while RUNNING.load(Ordering::Relaxed) {
+                let resp = rx.recv()?;
+
+                let src_addr = match resp.src_addr.ip() {
+                    IpAddr::V4(addr) => addr,
+                    IpAddr::V6(_) => return Err("IPv6 addresses are not supported".into()),
+                };
+
+                let dst_addr = match resp.dst_addr.ip() {
+                    IpAddr::V4(addr) => addr,
+                    IpAddr::V6(_) => return Err("IPv6 addresses are not supported".into()),
+                };
+
+                let v = generate_random_bytes(2)?;
+                let id = u16::from_ne_bytes([v[0], v[1]]);
+
+                // build response IP packet
+                use packet::Builder;
+                let ip_packet = packet::ip::v4::Builder::default()
+                    .id(id)?
+                    .ttl(64)?
+                    .source(src_addr)?
+                    .destination(dst_addr)?
+                    .udp()?
+                    .source(resp.src_addr.port())?
+                    .destination(resp.dst_addr.port())?
+                    .payload(&resp.data)?
+                    .build()?;
+
+                // // The following code will be better than above, the `ipv4_udp_build` function link is
+                // //
+                // // https://github.com/pysrc/study-udp/blob/59d7ba210a022d207c60ad5370de37110fefaefb/src/protocol.rs#L157-L252
+                // //
+                // let mut ip_packet = vec![0u8; 28 + resp.data.len()];
+                // protocol::ipv4_udp_build(
+                //     &mut ip_packet,
+                //     &src_addr.octets(),
+                //     resp.src_addr.port(),
+                //     &dst_addr.octets(),
+                //     resp.dst_addr.port(),
+                //     &resp.data,
+                // );
+
+                let mut write_pack = writer_session.allocate_send_packet(ip_packet.len() as u16)?;
+                write_pack.bytes_mut().copy_from_slice(ip_packet.as_ref());
+
+                // Send the response packet
+                writer_session.send_packet(write_pack);
+            }
+            Ok::<(), Box<dyn std::error::Error>>(())
+        };
+        if let Err(err) = block() {
+            println!("Writer {}", err);
+        }
+    });
+
```

Comment:
Add a print here telling the user that enter will end the session

Reply:
OK